operator/iamkeycontroller/observe.go

```diff
@@ -49,11 +49,9 @@ func (p *IAMKeyPipeline) Observe(ctx context.Context, mg resource.Managed) (mana
 		).RunWithContext(pctx)
 
 	if result != nil {
-		connDetails, err := toConnectionDetails(pctx.iamExoscaleKey)
```

Comment:
You skip to connection details altogether, isn't it necessary in some cases?

Reply:
The problem is that the pipeline fetches the credentials secret and checks the secret, if that fails (because e.g. the IAM key was just deleted), it doesn't make sense to try to fetch the connection details, as they might be already gone too.

Reply:
Then I'm ok with it.